lib/cable_ready/operation_builder.rb

```diff
@@ -24,10 +24,26 @@ def initialize(identifier)
     def add_operation_method(name)
       return if respond_to?(name)
       singleton_class.public_send :define_method, name, ->(*args) {
-        selector, options = nil, args.first || {} # 1 or 0 params
-        selector, options = options, {} unless options.is_a?(Hash) # swap if only selector provided
-        selector, options = args[0, 2] if args.many? # 2 or more params
-        options.stringify_keys!
+        if args.one? && args.first.respond_to?(:to_operation_options) && [Array, Hash].include?(args.first.to_operation_options.class)
+          case args.first.to_operation_options.class.name
+          when "Array"
+            selector, options = nil, args.first.to_operation_options
+              .select { |e| e.is_a?(Symbol) && args.first.respond_to?("to_#{e}".to_sym) }
+              .each_with_object({}) { |option, memo| memo[option.to_s] = args.first.send("to_#{option}".to_sym) }
+          when "Hash"
+            selector, options = nil, args.first.to_operation_options
+              .select { |e| e.is_a?(Symbol) }
+              .each_with_object({}) { |option, memo| memo[option[0]] = option[1] }
```

Comment:
Shouldn't handling Hash look more like this?

```ruby
when Hash
  selector, options = nil, args.first.to_operation_options
```

No need to reduce when it's already a Hash. Right?

Reply:
I think that you're right! If they pass a complex object as a key, that's on them. (And they could do that with the standard operation so it's not any better/worse.) Good catch.